src/backend/joanie/core/serializers/client.py

```diff
@@ -321,98 +318,95 @@ class Meta:
+        model = models.CourseRun
         fields = [
-            "code",
-            "course_runs",
-            "is_graded",
-            "organizations",
-            "position",
+            "end",
+            "enrollment_end",
+            "enrollment_start",
+            "id",
+            "resource_link",
+            "start",
             "title",
+            "state",
         ]
         read_only_fields = [
-            "code",
-            "course_runs",
-            "is_graded",
-            "organizations",
-            "position",
+            "end",
+            "enrollment_end",
+            "enrollment_start",
+            "id",
+            "resource_link",
+            "start",
             "title",
+            "state",
         ]
 
-    @property
-    def context_resource(self):
-        """
-        Retrieve the product/order resource provided in context.
-        If no product/order is provided, it raises a ValidationError.
-        """
-        try:
-            resource = self.context["resource"]
-        except KeyError as exception:
-            raise serializers.ValidationError(
-                'TargetCourseSerializer context must contain a "resource" property.'
-            ) from exception
 
-        if not isinstance(resource, (models.Order, models.Product)):
-            raise serializers.ValidationError(
-                "TargetCourseSerializer context resource property must be instance of "
-                "Product or Order."
-            )
+class AbstractTargetCourseRelationSerializer(serializers.ModelSerializer):
+    """
+    Abstract serializer for target course relations. Both Order and Product models have
+    those kind of relations.
+    """
 
-        return resource
+    position = serializers.IntegerField(read_only=True)
+    is_graded = serializers.BooleanField(read_only=True)
+    course_runs = serializers.SerializerMethodField("get_course_runs")
+    title = serializers.SerializerMethodField()
+    code = serializers.SerializerMethodField()
 
-    def get_target_course_relation(self, target_course):
-        """
-        Return the relevant target course relation depending on whether the resource context
-        is a product or an order.
-        """
-        if isinstance(self.context_resource, models.Order):
-            return target_course.order_relations.get(order=self.context_resource)
+    class Meta:
+        fields = ("code", "course_runs", "is_graded", "position", "title")
+        read_only_fields = fields
+
+    def get_course_runs(self, relation):
+        """Return targeted course runs for the given relation."""
+        if (
+            self.Meta.model  # pylint: disable=no-member
+            == models.OrderTargetCourseRelation
+        ):
+            model = relation.order
+        elif (
+            self.Meta.model  # pylint: disable=no-member
+            == models.ProductTargetCourseRelation
+        ):  # pylint: disable=no-member
+            model = relation.product
```

Comment:
I think I'm more confortable having 2 separate serializers like it's done here for example: https://github.com/openfun/marsha/blob/master/src/backend/marsha/core/permissions/organization.py#L209

The advantage is that we get a class name that is self describing.

Reply:
Oh, and you are deriving from this class below to split in 2 classes so one more reason to do it there!